conan/cli/commands/graph.py

```diff
@@ -298,3 +300,110 @@ def graph_explain(conan_api, parser,  subparser, *args):
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile,
+                                               conanfile_path=path,
+                                               cwd=cwd,
+                                               partial=args.lockfile_partial,
+                                               overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+
+    if path:
+        deps_graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
+                                                         args.user, args.channel,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update,
+                                                         check_updates=args.check_updates,
+                                                         is_build_require=args.build_require)
+    else:
+        deps_graph = conan_api.graph.load_graph_requires(args.requires, args.tool_requires,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update,
+                                                         check_updates=args.check_updates)
+    print_graph_basic(deps_graph)
+    ConanOutput().title("Checking remotes")
+
+    # Data structure to store info per library
+    dict_nodes = {}
+    dependencies = deps_graph.nodes[1:]
+    for node in dependencies:
+        dict_nodes.setdefault(node.name, {"cache_refs": set(), "version_ranges": [],
+                                          "latest_remote": None})["cache_refs"].add(node.ref)
+
+    for version_range in deps_graph.resolved_ranges.keys():
+        dict_nodes[version_range.name]["version_ranges"].append(version_range)
+
+    # find in remotes
+    _find_in_remotes(conan_api, dict_nodes, remotes)
+
+    #Filter nodes with no outdated versions
+    filtered_nodes = {}
+    for node_name, node in dict_nodes.items():
+        if node['latest_remote'] is not None and sorted(list(node['cache_refs']))[0] < \
+            node['latest_remote']['ref']:
+            filtered_nodes[node_name] = node
+
+    return filtered_nodes
+
+
+def _find_in_remotes(conan_api, dict_nodes, remotes):
+    for node_name, node_info in dict_nodes.items():
+        ref_pattern = ListPattern(node_name, rrev=None, prev=None)
+        for remote in remotes:
+            try:
+                remote_ref_list = conan_api.list.select(ref_pattern, package_query=None,
+                                                        remote=remote)
+                if not remote_ref_list.recipes:
+                    continue
+            except ConanException as message:
+                ConanOutput().warning(f"{message} in remote {remote.name}")
+                continue
```

Comment:
When does this fail? Should it be a warning in all cases?
It is not the same that the remote is unavailable => should raise an error, that the package is not in the remote, which is ok, and shouldn't even warn

Reply:
Is it okey to remove the message then? We need to catch the exception when it doesn't find anything but maybe the continue is enough.

Do we want to additionally handle the error you mention when it can't connect to the remote?

Reply:
yes, you need to treat the different errors differently. Connection errors are bad, should be raised. 404-not found are not really an error in this context, not even a warning.